egress/src/filter.rs

```diff
@@ -0,0 +1,1415 @@
+    for list in filter.values_mut() {
+        list.sort_by_key(|item| (item.from));
+    }
+    eprintln!("length filter: {:?}", filter.len());
+    Ok(filter)
+}
+
+pub async fn get_filter(
+    conn: &PooledPgConn<'_>,
+    from: DateTime<Utc>,
+    to: DateTime<Utc>,
+    filter_table: FilterTimeseriesTableLock,
+    label: FilterLabel,
+) -> Result<Option<Vec<FilterData>>, tokio_postgres::Error> {
+    // get the background filter list, and lookup this label
+    // have to clone, otherwise does not work...
+    let ft = filter_table
+        .read()
+        .map_err(|e| Error::Lock(e.to_string()))
+        .unwrap()
+        .clone();
+
+    let filter = ft.get(&label);
+    // TODO: if the label has none for sensor / level should it match on all???
+    // create a structure to keep what is applicable
+    let mut applicable_ts: Vec<(i64, DateTime<Utc>, DateTime<Utc>)> = vec![];
+    // fill the structure
+    match filter {
+        Some(options) => {
+            for opt in options {
+                //    // is this applicable?
+                let ft_t = timerange_overlap(
+                    Timerange {
+                        from: Some(from),
+                        to: Some(to),
+                    },
+                    Timerange {
+                        from: Some(opt.from),
+                        to: opt.to,
+                    },
+                );
+                // have overlap
+                if let Some(times) = ft_t {
+                    if let Some(tt) = times.to {
+                        applicable_ts.push((opt.tsid, times.from.unwrap(), tt));
+                    } else {
+                        // open ended (to time from request)
+                        applicable_ts.push((opt.tsid, times.from.unwrap(), to));
+                    }
+                }
+            }
+            if applicable_ts.is_empty() {
+                return Ok(None); // no applicable time series for this time range?
+            }
+        }
+        None => return Ok(None), // no prioritized timeseries
+    }
+
+    let mut futures = applicable_ts
+        .iter()
+        .map(|(tsid, from, to)| async move {
+            let get_ts = format!(
+                "SELECT obsvalue, obstime, timeseries FROM data WHERE (timeseries = {tsid} \
+                    AND obstime >= '{from}' AND obstime < '{to}')",
+            );
+            conn.query(&get_ts, &[]).await
+        })
+        .collect::<FuturesOrdered<_>>()
+        .enumerate();
+
+    let mut fails: Vec<usize> = Vec::new();
+    let mut data = Vec::with_capacity(applicable_ts.len()); // best guess for size, but could be too big?
+
+    while let Some((i, res)) = futures.next().await {
+        let rows = match res {
+            Ok(val) => val,
+            Err(_err) => {
+                fails.push(i);
+                continue;
+            }
+        };
```

Comment:
Should we do something with the failed queries? Doesn't look that useful right now

Reply:
Oops I missed this comment and repeated it in my review, I guess I'll resolve that one. If you want to log it I think you should just log it now. I guess we don't have logging set up in egress, but we can just copy the code. If you don't want to deal with it now, I think you should leave a TODO behind so the intention is clear for the next person who sees this.